test/ProductConstructionService.DependencyFlow.Tests/SubscriptionOrPullRequestUpdaterTests.cs

```diff
@@ -45,13 +48,18 @@ protected override void RegisterServices(IServiceCollection services)
 
     protected override async Task BeforeExecute(IServiceProvider context)
     {
-        var dbContext = context.GetRequiredService<BuildAssetRegistryContext>();
-        foreach (Action<BuildAssetRegistryContext> update in ContextUpdates)
+        if (!initDone)
```

Comment:
We need this for when we call `ProcessPendingUpdates` more than once, since the way it's configured, it will try to add all builds, tests, etc.. to the DB again

Reply:
I don't follow. This sounds like it needs to be only called once? How does it get called multiple times with your changes?

If this is about calling `WhenProcessPendingUpdatesAsyncIsCalled` multiple times, I suspect you might be using it wrong or not how it was intended. Then maybe it's better to change the functionality (e.g. support multiple build parameters wrapped in the `Execute`) rather than putting more state and branching inside.

Reply:
Also please consider every time you need to add explanation into a PR as a comment whether it shouldn't have been a comment in the code. Usually the answer is yes. Here it will get lost quick.